### ISSUE TITLE
1.2.15 works perfectly, 1.2.16 Unrecognized configuration section, 1.2.17 no error but all values 0

### ISSUE THREAD

**Issue by dougsydney** [erfg12/memory.dll#48]
Everything in my project kept exactly the same but only updating the version of memory.dll I'm facing issues:

As per the title:

1.2.15 - works perfectly
1.2.16 - with any "Get" memory method I get an internal exception "ConfigurationErrorsException: Unrecognized configuration section requestedExecutionLevel."
1.2.17 - I get no error but all values returned are 0. IsAdmin is true which tells me admin privs is working fine.

I'm running with admin privs and as mentioned the only thing that's changing in my project is the memory.dll version. Basically I just want to use the new overload of AoBScan to return more than one address.

What could be the problem? Thanks

**Comment by thgiang** [comment]
I see the same error when run in x64. Try to run probject on any CPU will fix my problem.

**Comment by erfg12** [comment]
Did you adjust the target framework?

**Comment by thgiang** [comment]
This version add Aob Signature scan. It have to run on 4.7.1 NET Framework  or higher (4.7.2)

**Comment by dougsydney** [comment]
Thanks for your replies. Yes running "Any CPU" already and .NET framework version is targeted correctly (I used 4.5 with 1.2.15 but am forced to bump up to 4.7.1 or 4.7.2 when using the new version). I tested using an older version of Memory.dll with the latest .NET framework version and it worked great.

I managed to solve it:

I created a whole new project with the latest .NET version just to test and same issue. I notice however in the debugger that when it isn't working the mLib library has 0 modules and no pHandle so it's not even able to hook into the exe. 

I've gone through the commits binary search styles and compiled each and tested and found the issue starts occurring with this commit: https://github.com/erfg12/memory.dll/commit/bc9a557e9f62b07f393bc265ec5646245a1d14fb

The refactoring of this method screwed up how I hook into the process using this code


```
  int gameProcId = mLib.getProcIDFromName("myexe.exe");

  if (gameProcId != 0)
  {
         mLib.OpenProcess(gameProcId.ToString());
  }
```
The tooltip for OpenProcess says it accepts either a process name or a process ID but if you look at the source it's clear it only is checking process names so with the process ID it fails to open. In 1.2.15 it worked.

<img width="635" alt="process" src="https://user-images.githubusercontent.com/45820093/49934709-f85c1780-ff33-11e8-9159-9d2f285ea645.png">

Why I go about loading in that odd way you may wonder and which also leads to a second bug is because I recall it's due to the fact OpenProcess would return false for me despite the hook working so I had to first check for ProcessID instead. I delved deeper and it seems the error I got with 1.2.16 is infact related to this. After debugging I found what was happening is that the first time Debug.WriteLine was called (these are scattered around in Memory.dll) it would cause that nasty "Unrecongized configuration section" exception, but the subsuqnet times Debug.WriteLine is called it works fine. You can see with 1.2.16 the exception is uncaught when hitting Debug.WriteLine for the first time:

<img width="1342" alt="error" src="https://user-images.githubusercontent.com/45820093/49934722-01e57f80-ff34-11e8-8eac-faa740181d4b.png">


In 1.2.16 that happened somewhere where the exception was unhandled so I saw it but with 1.2.17 it was caught so everything looked okay apart from the 0 values. In OpenProcess the very last call is a Debug.WriteLine method to say it successfully hooked, that would cause that exception and the method would return false even when working and therefore the need for the round about code above.

I tried messing around with app.config and that exception managed to morph into "Configuration system failed to initialize" but I couldn't get rid of it. Again it's only the first Debug.WriteLine that causes this, and if that's caught after that it works fine. The issue also occurs when using Debug.WriteLine in my own application so it's not to do with Memory.dll I assume it's just some strange thing with how .NET is installed on this machine or something.

So to sum up fixing GetProcess to actually allow loading from processId would be great as that was the main cause of the issue. Thanks for your time.

**Comment by hollow87** [comment]
If you would use the process ID as an integer rather than string yes it would open it
https://github.com/erfg12/memory.dll/blob/81ad14545c929757924f084a85309ffd964f4512/Memory/Class1.cs#L262-L317

Yes it looks like the XML comments do not match the defined behavior the string overload should only take process name not process ID anymore.

Change your code to either use the process name or get rid of the .ToString() on the integer.
`mLib.OpenProcess("myexe.exe");`
or
`mLib.OpenProcess(gameProcId);`

The string overload for OpenProcess takes the name and gets the process ID from the name then calls the overload that just takes an integer (the process ID)

That Exception your getting in regards to Debug.WriteLine I would check your applications app.config for the section its asking Also make sure your running the application as Administrator.

**Comment by thgiang** [comment]
This code part work for me:
`using Memory;
Mem MemLib = new Mem();

**MemLib.OpenProcess(MyGameProcess.Id);**
MemLib.readFloat("0x123456");
`
"0x123456" is in STRING

**Comment by erfg12** [comment]
If you have questions on how a function works, ask in our Discord.
